Automattic/jetpack
# Jetpack Boost: Add uPlot score history graph

- Fixes https://github.com/Automattic/boost-cloud/issues/284

## Proposed changes:


It adds a Historical Performance section with a chart to Jetpack Boost page `/wp-admin/admin.php?page=jetpack-boost`. You need to be Boost premium to load some data.

* Add `uPlot` library to `package.json`
* Add new components `BoostScoreGraph`, `PerformanceHistory`
* Add new storybook stories
* Integrate `PerformanceHistory` with svelte to show it in Jetpack Boost page.
* Add jetpack endpoint `/wp-json/jetpack-boost/v1/speed-scores-history`

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:

It's necessary to have purchased a site with Jetpack Boost premium a few days ago. Or you will need to override the data returned by the endpoint.

* Install Jetpack Beta on your site
* Set this branch `add/uplot-boost-graph` for Jetpack Boost.

<img width="713" alt="Screenshot 2023-08-18 at 14 37 34" src="https://github.com/Automattic/jetpack/assets/779993/e9fd071d-f0f7-4914-8033-82f3e5b92765">

* Go to `Jetpack > Boost` - `/wp-admin/admin.php?page=jetpack-boost`
* Active the premium plan. The corn runs daily for all premium sites, if you just purchased the premium plan, you will need to wait to see data from the API. 
* Observe the new graph.
* Observe a network request to `/wp-json/jetpack-boost/v1/speed-scores-history`







**Storybook**
* Execute the storybook `cd projects/js-packages/storybook && npm run storybook:dev`
* Access http://localhost:50240/?path=/docs/plugins-boost-performance-history--docs
* Observe the new `PerformanceHistory` component

## Screenshot

![jetpack-boost-chart](https://github.com/Automattic/jetpack/assets/779993/5657e958-2acd-42ad-99fe-c463f3b38db9)



Storybook:
![jetpack-boost-with-tooltip](https://github.com/Automattic/jetpack/assets/779993/e4812002-fb8f-4d38-a4c1-7037ac5fc1e8)
